dev/integration_tests/native_driver_test/android/app/src/main/kotlin/com/example/native_driver_test/extensions/NativeDriverSupportPlugin.kt

```diff
@@ -0,0 +1,108 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+@file:Suppress("PackageName")
+
+package com.example.native_driver_test.extensions
+
+import android.app.Activity
+import android.os.SystemClock
+import android.view.MotionEvent
+import io.flutter.Log
+import io.flutter.embedding.engine.plugins.FlutterPlugin
+import io.flutter.embedding.engine.plugins.activity.ActivityAware
+import io.flutter.embedding.engine.plugins.activity.ActivityPluginBinding
+import io.flutter.plugin.common.MethodCall
+import io.flutter.plugin.common.MethodChannel
+import io.flutter.plugin.common.MethodChannel.MethodCallHandler
+
+class NativeDriverSupportPlugin :
+    ActivityAware,
+    FlutterPlugin,
+    MethodCallHandler {
+    private val tag = "NativeDriverSupportPlugin"
+    private lateinit var channel: MethodChannel
+    private var activity: Activity? = null
+
+    override fun onAttachedToEngine(binding: FlutterPlugin.FlutterPluginBinding) {
+        channel = MethodChannel(binding.binaryMessenger, "native_driver")
+        channel.setMethodCallHandler(this)
+    }
+
+    override fun onDetachedFromEngine(binding: FlutterPlugin.FlutterPluginBinding) {
+        channel.setMethodCallHandler(null)
+    }
+
+    override fun onMethodCall(
+        call: MethodCall,
+        result: MethodChannel.Result
+    ) {
+        val activity = this.activity
+        if (activity == null) {
+            Log.w(tag, "Received method channel, but no current activity")
+            return
+        }
+        when (call.method) {
+            "ping" -> {
+                result.success(null)
+            }
+            "tap_view" -> {
+                // Decode the selector.
+                val kind = call.argument<String>("kind")
+                lateinit var selector: NativeSelector
+                when (kind) {
+                    "byNativeAccessibilityLabel" -> {
+                        selector = NativeSelector.ByContentDescription(call.argument("label")!!)
+                    }
+                    "byNativeIntegerId" -> {
+                        selector = NativeSelector.ByViewId(call.argument("id")!!)
+                    }
+                    else -> {
+                        result.error("INVALID_SELECTOR", "Not supported", kind)
```

Comment:
Does this need to return to avoid continuing to execute?

Reply:
Done!